meta-balena-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -235,6 +296,10 @@ if type secureboot_setup >/dev/null 2>&1 && secureboot_setup; then
 fi
 
 if [ "$CRYPT" = "1" ]; then
+    # If we are going for the encryption, first of all verify that the image
+    # we are about to flash is correctly signed.
+    verify_image_signature "${BALENA_IMAGE}"
```

Comment:
paranoia: maybe verify the return code and fail if false, even though it's probably not needed?